### PR TITLE
Implement Pipfile and remove extra requirements.txt files

### PR DESCRIPTION
**Pull request by gtmanfred**
### What does this PR do?
Instead of having extra requirements.txt files, we can just specify only install dependencies based on which version of python we are using.

Also, add a Pipfile for pipenv, because it is super awesome, and gives us lockfiles/virtualenvs.

Based on what we saw at PyCon 2018, this will be a super awesome way to manage pip stuff and will be included in pip in the future.

https://us.pycon.org/2018/schedule/presentation/243/

### Commits signed with GPG?

Yes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,6 +11,7 @@ MANIFEST
 *.wpr
 *.wpu
 *.DS_Store
+Pipfile.lock
```
Comment (terminalmage) [inline .gitignore]: The creator of pipenv recommended keeping the Pipfile and lockfile both in version control.

Comment (gtmanfred) [reply .gitignore]: Agreed, once I learned that you can still install newer stuff even if the lock file is there, it makes sense that you should put the lock file in VCS.

**Comment by gtmanfred** [comment]
Thanks @kennethreitz for pipenv

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+[dev-packages]
+mock = ">=2.0.0"
+apache-libcloud = ">=0.14.0"
+boto = ">=2.32.1"
+boto3 = ">=1.2.1"
+moto = ">=0.3.6"
+SaltPyLint = ">=v2017.3.6"
+pytest = ">=3.5.0"
+Pytest-catchlog = {git = "git://github.com/eisensheng/pytest-catchlog.git", ref = "develop", markers = "python_version < '3.0'"}
```
Comment (s0undt3ch) [inline Pipfile]: Drop this one, if it `py.test` explodes, I have a fix

File: requirements/dev.txt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+-r base.txt
+
+mock>=2.0.0
+apache-libcloud>=0.14.0
+boto>=2.32.1
+boto3>=1.2.1
+moto>=0.3.6
+SaltPyLint>=v2017.3.6
+pytest>=3.5.0
+git+https://github.com/eisensheng/pytest-catchlog.git@develop#egg=Pytest-catchlog; python_version < '3.0'
```
Comment (s0undt3ch) [inline requirements/dev.txt]: same here

**Review by s0undt3ch — APPROVED** [review]
:tada:

File: Pipfile (hunk 2 above)
Comment (gtmanfred) [reply Pipfile]: Done

File: requirements/dev.txt (hunk 3 above)
Comment (gtmanfred) [reply requirements/dev.txt]: Done

Comment (gtmanfred) [reply requirements/dev.txt]: looks like pytest still works.

**Comment by kennethreitz** [comment]
This is awesome! ✨🍰✨

File: requirements/dev.txt (hunk 3 above)
Comment (s0undt3ch) [reply requirements/dev.txt]: :applause:

File: tests/conftest.py [hunk 4]
```diff
@@ -102,6 +102,14 @@ def pytest_addoption(parser):
              'SSH server on your machine. In certain environments, this '
              'may be insecure! Default: False'
     )
+    test_selection_group.addoption(
```
Comment (rallytime) [inline tests/conftest.py]: Is this change intentional? This seems unrelated to the rest of the changes here.

File: requirements/dev_python27.txt [hunk 5]
```diff
@@ -1,11 +1,2 @@
...
-mock>=2.0.0
-apache-libcloud>=0.14.0
-boto>=2.32.1
-boto3>=1.2.1
-moto>=0.3.6
-SaltPyLint>=v2017.3.6
-pytest>=3.5.0
-git+https://github.com/eisensheng/pytest-catchlog.git@develop#egg=Pytest-catchlog
-git+https://github.com/saltstack/pytest-salt.git@master#egg=pytest-salt
+# This is for legacy purposes
```
Comment (rallytime) [inline requirements/dev_python27.txt]: I know this is picky, but it might make sense for this comment to match the one in `dev_python34.txt` (and I liked that one the best of the two). 😄

File: requirements/dev_python27.txt [hunk 6]
```diff
@@ -1,11 +1,2 @@
--r base-py2.txt
-
-mock>=2.0.0
-apache-libcloud>=0.14.0
-boto>=2.32.1
-boto3>=1.2.1
-moto>=0.3.6
-SaltPyLint>=v2017.3.6
-pytest>=3.5.0
-git+https://github.com/eisensheng/pytest-catchlog.git@develop#egg=Pytest-catchlog
```
Comment (rallytime) [inline requirements/dev_python27.txt]: I don't see the pytest-catchlog in the new dev.txt file. I am not sure what that particular addition to the Py27 requirements file is for since there's no comment about it, but I wanted to double check that this was intentional to leave that out.

**Review by rallytime — COMMENTED** [review]
This is neat! I have a couple of very picky suggestions and a question on this change.

File: tests/conftest.py (hunk 4 above)
Comment (gtmanfred) [reply tests/conftest.py]: It was intentional, it fixes pytest for salt, so you can run integration tests.

I can move it to a new PR if you want.

File: requirements/dev_python27.txt (hunk 5 above)
Comment (gtmanfred) [reply requirements/dev_python27.txt]: Will do

File: requirements/dev_python27.txt (hunk 6 above)
Comment (gtmanfred) [reply requirements/dev_python27.txt]: pytest-catchlog has been merged into pytest upstream, so we no longer needed it specified.

**Review by s0undt3ch — APPROVED** [review]

File: requirements/dev_python27.txt (hunk 6 above)
Comment (rallytime) [reply requirements/dev_python27.txt]: Perfect!

File: tests/conftest.py (hunk 4 above)
Comment (rallytime) [reply tests/conftest.py]: Nah, it's good. Just making sure.

**Review by rallytime — APPROVED** [review]
